### ISSUE TITLE
dynamic pvc to create resources with predictable names on azure files

### ISSUE THREAD

**Issue by dzmitry-lahoda** [Azure/AKS#2885]
**What happened**:

I claimed dynamic `pvc`, and got allocation of `pv` named `pvc-faf0f176-8b8d-11e8-923b-deb28c58d242` and relevant Azure Files resources.

**What you expected to happen**:

I want to be able to add annotation to `pvc`, asking to add prefix to `pv` name. 

If I want prefix like, `my-super-box`, I annotate `pvc` with it. 

So that when azure file resources is created, is see `my-super-box-pvc-faf0f176-8b8d-11e8-923b-deb28c58d242`

**How to reproduce it (as minimally and precisely as possible)**:

Create dynamic `pvc` which creates Azure Files resource.

**Anything else we need to know?**:

Alternatives are:
- build custom portal board to map pods mounting pv mounting pv to able to find azure file used by pod. pods are predictable name prefixed already. 
- automate creation of static pvc from named static pv, but that is not so good solution and needs amount of work which may be comparable to make it into aks with dynamic pvc

**Environment**:
- Kubernetes version (use `kubectl version`): 1.21
- General description of workloads in the cluster: game servers

**Comment by dzmitry-lahoda** [comment]
something like this?

```
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: shared-persistent-fs
provisioner: file.csi.azure.com
allowVolumeExpansion: true
parameters:
  skuName: Standard_LRS
  fileSharePrefix: my-super-box
```

**Comment by andyzhangx** [comment]
> something like this?
> 
> ```
> kind: StorageClass
> apiVersion: storage.k8s.io/v1
> metadata:
>   name: shared-persistent-fs
> provisioner: file.csi.azure.com
> allowVolumeExpansion: true
> parameters:
>   skuName: Standard_LRS
>   fileSharePrefix: my-super-box
> ```

yes, does that work for you?

**Comment by andyzhangx** [comment]
https://github.com/kubernetes-sigs/azurefile-csi-driver/pull/966 would fix this issue, and this feature would be in v1.14.0 on AKS

**Comment by andyzhangx** [comment]
would be fixed with v1.14.0 release: https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.14.0
